cmd/ax/query.go

```diff
@@ -80,6 +91,74 @@ func buildFilters(wheres []string) []common.QueryFilter {
 	return filters
 }
 
+var membershipFilterRegex = regexp.MustCompile(`([^!=<>]+)\s*:\s*(.*)`)
+
+func buildMembershipFilters(oneOfs []string, notOneOfs []string) []common.MembershipFilter {
+	// Build a nested map of field names and their membership constraints
+	fields := make(map[string]map[bool][]string)
+	for _, oneOfClause := range oneOfs {
+		matches := membershipFilterRegex.FindAllStringSubmatch(oneOfClause, -1)
+		if len(matches) != 1 {
+			fmt.Println("Invalid one-of clause", oneOfClause)
+			os.Exit(1)
+		}
+		fieldName, value := matches[0][1], matches[0][2]
+		field, ok := fields[fieldName]
+		if !ok {
+			fields[fieldName] = map[bool][]string{
+				true:  {value},
+				false: {},
+			}
+		} else {
+			field[true] = append(fields[fieldName][true], value)
+		}
+	}
+	for _, notOneOfClause := range notOneOfs {
```

Comment:
This really feels like the same code twice, can this not be factored out somehow?

Reply:
Extracted this into another function.